java-checks-test-sources/default/src/main/java/checks/ConfigurationBeanNamesCheckSample.java

```diff
@@ -0,0 +1,80 @@
+package checks;
+
+import javax.annotation.Nullable;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.userdetails.User;
+
+public class ConfigurationBeanNamesCheckSample {
+
+  @Configuration
+  class Config1 {
+    @Bean
+    public User user() {
```

Comment:
I think the usage of `User` instead of a simple dummy class (like `foo`, or an own `class User{}`) overly complicates the test, because:
1. lots of boilerplate code for the constructor call, but the check does not even enter method bodies, so it's not necessary for the test
2. it creates an otherwise unnecessary import